system/modules/isotope/library/Isotope/Frontend.php

```diff
@@ -828,4 +830,29 @@ public static function getIsotopeMessages()
     {
         return Message::generate();
     }
+
+    /**
+     * Switches the environment to the given language.
+     * 
+     * @param string $language
+     */
+    private static function setLanguage($language)
+    {
+        $GLOBALS['TL_LANGUAGE'] = $language;
+
+        if (class_exists(ContaoCoreBundle::class)) {
+            /** @var ContainerInterface $container */
+            $container = System::getContainer();
+
+            if ($container->has('request_stack') && null !== ($request = $container->get('request_stack')->getCurrentRequest())) {
+                $request->setLocale($language);
+            }
+
+            if ($container->has('translator')) {
+                $container->get('translator')->setLocale($language);
+            }
+        }
+
+        System::loadLanguageFile('default', $language);
```

Comment:
I have removed bypassing the static cache, I think that's not really necessary.

Reply:
I don't think that's correct. If you load english, then german and then want english again, you must use the `$blnNoCache` flag.

Reply:
Why? The languages are stored in separate static caches.

Reply:
not sure what you mean, they are loaded in a global array? So if you load english then german and then english again, german would remain in the global array.

Reply:
Yeah, I misunderstood the cache check in `System::loadLanguageFile`. I am now wondering though if the helper function should always force load the language, since that might degrade performance. I am not familiar on whether theses function are called in the front end multiple times as well?

Reply:
I think they are only loaded when printing a PDF or sending a notification. It _might_ be loaded multiple times, but then we also need to make sure the language is correct. Unfortunately, it seems we can't know the current language (to not load if not necessary)

Reply:
> Unfortunately, it seems we can't know the current language (to not load if not necessary)

Hm, why not, I think we can do an early out with

```php
if ($GLOBALS['TL_LANGUAGE'] ?? null === $language) {
    return;
}
```

`$GLOBALS['TL_LANGUAGE']` should always be set to the actual language.

Reply:
`System::loadLanguageFile()` does not set `$GLOBALS['TL_LANGUAGE']`, so why would that work?

Reply:
Within Isotope's own code base, `System::loadLanguageFile` with a language override is not executed without also setting the language in `$GLOBALS['TL_LANGUAGE']`. And if another extension does this, you would have wrong translations in other places for this request anyway, so I think it is fair to assume that the currently loaded language for translations is the same language as the one set for the current request.

Reply:
Maybe you can try to explain why you're trying to change the cache behaviour? It currently works and any change _might_ worsen it but not really improve anything?

Reply:
The current PR fixes the issue of the translator not using the correct languages and otherwise does not change the behaviour - except for the cache parameter for `System::loadLanguageFile`, which I will revert again. The current discussion is just about whether or not we can return early in the new helper method, but it's not really that important anyway.